pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/FunctionConfigUtils.java

```diff
@@ -317,9 +319,15 @@ public static FunctionConfig convertFromDetails(FunctionDetails functionDetails)
         }
         if (functionDetails.getSource().getSubscriptionType() == Function.SubscriptionType.FAILOVER) {
```

Comment:
This whole conditional statement doesn't see right to me.  Why are se setting the processing guarantee by looking at what the subscription is?  We already setting the processing guarantee in a previous line:

https://github.com/apache/pulsar/pull/7647/files#diff-239e167c18ea8591c85432948ab5040aL297